crates/symbolicator/src/services/symbolication.rs

```diff
@@ -1267,20 +1267,16 @@ fn symbolicate_stacktrace(
     metrics: &mut StacktraceMetrics,
     signal: Option<Signal>,
 ) -> CompleteStacktrace {
-    let mut stacktrace = CompleteStacktrace {
-        thread_id: thread.thread_id,
-        is_requesting: thread.is_requesting,
-        registers: thread.registers.clone(),
-        frames: vec![],
-    };
+    let mut frames = vec![];
+    let mut frames_iter = thread.frames.into_iter().enumerate().peekable();
 
-    for (index, mut frame) in thread.frames.into_iter().enumerate() {
+    while let Some((index, mut frame)) = frames_iter.next() {
```

Comment:
I was sure that clippy would complain about this being equivalent to for (I realize that for doesn't work here due to borrowing).

Reply:
It did complain (as expected), but putting in the `.peek()` made it realize that we do need a mutable `frames_iter` to be around.

Reply:
Clippy really is quite amazing.